### PR TITLE
✨ Instance deletion logs

### LINKED ISSUE #1139: Debugging hanging deletes is hard
/kind bug

It is hard, given the error logs today, to understand what is happening when a Cluster fails to delete. CAPA/CAPI/CABPK do not provide insight into what is hanging when it gets deleted.

I want to know what **exactly** is trying to be deleted. If it's hanging, the last log I should see is "trying to delete ..." or similar. 

If we were able to understand why deletes are hanging then we could provide an even more useful error message. Perhaps we can add a timeout to the delete that provides the user with action if the timeout is reached.

/priority important-soon
/milestone v0.4.x

### PR DESCRIPTION
**Pull request by randomvariable**
**What this PR does / why we need it**:
Cleans up logging during deletes, and adds documentation on events.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1139 

Just need to do a sanity check before removing [wip]

### CODE REVIEW

**Comment by ncdc** [comment]
@randomvariable is this almost ready to go?

**Comment by randomvariable** [comment]
@ncdc Let you know in 61 mins when the ec2 instance goes away

**Comment by detiber** [comment]
/lgtm
lgtm pending final verification

**Comment by randomvariable** [comment]
I initiated the wrong set of actions in EC2 to invoke this bit of code, and need to start again :(

File: docs/observability.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Observability
+
+## Events
+
+The following events are published by this controller:
```
[the remark this answers is not in the document]
Comment (randomvariable) [reply docs/observability.md]: Done

**Comment by randomvariable** [comment]
Done. Have added a note on how to reach the bad state.

**Pull request merged**